dev/ToastNotifications/ToastNotificationUtility.cpp

```diff
@@ -0,0 +1,221 @@
+    }
+    else
+    {
+        return RetrieveUnpackagedAppId();
+    }
+}
+
+void RegisterAssets(std::wstring const& appId, winrt::Microsoft::Windows::ToastNotifications::ToastAssets const& activationInfo, wil::unique_cotaskmem_string const& clsid)
+{
+    wil::unique_hkey hKey;
+    // subKey: \Software\Classes\AppUserModelId\{AppGUID}
+    std::wstring subKey{ c_appIdentifierPath + appId };
+
+    THROW_IF_FAILED(RegCreateKeyEx(
+        HKEY_CURRENT_USER,
+        subKey.c_str(),
+        0,
+        nullptr /* lpClass */,
+        REG_OPTION_NON_VOLATILE,
+        KEY_ALL_ACCESS,
+        nullptr /* lpSecurityAttributes */,
+        &hKey,
+        nullptr /* lpdwDisposition */));
+
+    RegisterValue(hKey, L"DisplayName", reinterpret_cast<const BYTE*>(activationInfo.DisplayName().c_str()), REG_EXPAND_SZ, activationInfo.DisplayName().size() * sizeof(wchar_t));
+    RegisterValue(hKey, L"IconUri", reinterpret_cast<const BYTE*>(activationInfo.IconPath().AbsoluteUri().c_str()), REG_EXPAND_SZ, activationInfo.IconPath().AbsoluteUri().size() * sizeof(wchar_t));
+
+    std::wstring wideStringClsid{ clsid.get() };
+    RegisterValue(hKey, L"CustomActivator", reinterpret_cast<const BYTE*>(wideStringClsid.c_str()), REG_SZ, wideStringClsid.size() * sizeof(wchar_t));
+}
+
+void RegisterComServer(wil::unique_cotaskmem_string const& processName, wil::unique_cotaskmem_string const& clsid)
+{
+    wil::unique_hkey hKey;
+    //subKey: Software\Classes\CLSID\{comActivatorGuidString}\LocalServer32
+    std::wstring subKey{ c_clsIdPath + clsid.get() + LR"(\LocalServer32)" };
+
+    THROW_IF_FAILED(RegCreateKeyEx(
+        HKEY_CURRENT_USER,
+        subKey.c_str(),
+        0,
+        nullptr /* lpClass */,
+        REG_OPTION_NON_VOLATILE,
+        KEY_ALL_ACCESS,
+        nullptr /* lpSecurityAttributes */,
+        &hKey,
+        nullptr /* lpdwDisposition */));
+
+    std::wstring comRegistrationExeString{ c_quote + processName.get() + c_quote + c_toastActivatedArgument };
+
+    RegisterValue(hKey, nullptr, reinterpret_cast<const BYTE*>(comRegistrationExeString.c_str()), REG_SZ, (comRegistrationExeString.size() * sizeof(wchar_t)));
+}
+
+void UnRegisterComServer(std::wstring const& clsid)
+{
+    wil::unique_hkey hKey;
+    //subKey: Software\Classes\CLSID\{comActivatorGuidString}\LocalServer32
+    std::wstring subKey{ c_clsIdPath + clsid + LR"(\LocalServer32)" };
+
+    THROW_IF_FAILED(RegDeleteKeyEx(
+        HKEY_CURRENT_USER,
+        subKey.c_str(),
+        KEY_WOW64_64KEY,
+        0));
+}
+
+void UnRegisterAppIdentifierFromRegistry()
+{
+    std::wstring appIdentifier{ RetrieveAppId() };
+
+    wil::unique_hkey hKey;
+    //subKey: \Software\Classes\AppUserModelId\{AppGUID}
+    std::wstring subKey{ c_appIdentifierPath + appIdentifier };
+
+    THROW_IF_FAILED(RegDeleteKeyEx(
+        HKEY_CURRENT_USER,
+        subKey.c_str(),
+        KEY_WOW64_64KEY,
+        0));
+}
+
+HRESULT GetActivatorGuid(std::wstring& activatorGuid)
```

Comment:
Put this in a try catch return block. Methods either return HRESULT or they throw. They don't do both.